src/pages/identity/RegisterPage.tsx

```diff
@@ -5,47 +5,157 @@ import {
-          placeholder="email"
-          onIonChange={(e: any) => setEmail(e.target.value)}
-        ></IonInput>
-        <IonInput
-          type="password"
-          placeholder="password"
-          onIonChange={(e: any) => setPassword(e.target.value)}
-        ></IonInput>
-        <IonInput
-          type="password"
-          placeholder="confirm password"
-          onIonChange={(e: any) => setConfirmedPassword(e.target.value)}
-        ></IonInput>
-        <IonButton onClick={register}>Register</IonButton>
+        <h1>Welcome to Book Club App</h1>
+        <IonItem className={`${isNameValid === false && 'ion-invalid'} ${isNameTouched && 'ion-touched'}`}>
+          <IonLabel position="floating">User Name</IonLabel>
+          <IonInput onIonChange={onNameInput}></IonInput>
+          {isNameValid === false && (
+            <IonNote slot="error">please enter you name</IonNote>
+          )}
+        </IonItem>
 
-        <p>
-          {" "}
-          Already have an account?{" "}
-          <IonButton routerLink="/login">Login</IonButton>{" "}
-        </p>
+        <IonItem className={`${isEmailValid === false && 'ion-invalid'} ${isEmailTouched && 'ion-touched'}`}>
+          <IonLabel position="floating">Email Address</IonLabel>
+          <IonInput type="email" onIonChange={onEmailInput}></IonInput>
+          {isEmailValid === false && (
+            <IonNote slot="error">{invalidEmailError}</IonNote>
+          )}
+        </IonItem>
+
+        <IonItem className={`${isPasswordValid === false && 'ion-invalid'} ${isPasswordTouched && 'ion-touched'}`}>
+          <IonLabel position="floating">Password</IonLabel>
+          <IonInput type="password" onIonChange={onPasswordInput}></IonInput>
+          {isPasswordValid === false && (
+            <IonNote slot="error">min 6 characters</IonNote>
+          )}
+        </IonItem>
+
+        <IonItem className={`${isConfirmedPasswordValid === false && 'ion-invalid'} ${isConfirmedPasswordTouched && 'ion-touched'}`}>
+          <IonLabel position="floating">Confirm Password</IonLabel>
+          <IonInput type="password" onIonChange={onConfirmPasswordInput}></IonInput>
+          {isConfirmedPasswordValid === false && (
+            <IonNote slot="error">should match the password above</IonNote>
+          )}
+        </IonItem>
+        <IonItem lines="none">
+          <IonCheckbox slot="start" onIonChange={onTermsCheck}></IonCheckbox>
+          <IonLabel color={`${isTermsCheckedValid === false && 'danger'}`}>
+            I have taken note of the
+            <br />
+            data protection regulations
+          </IonLabel>
+        </IonItem>
+        <IonButton onClick={register}>Register</IonButton>
+        <IonItem lines="none">
+          <p>
+            Already have an account?
+            <IonRouterLink routerDirection="forward" routerLink="/login"> Log in</IonRouterLink>
+          </p>
+        </IonItem>
       </IonContent>
     </IonPage>
   );
```

Comment:
This needs rework! It is dangerous to have this many useState hooks as it always triggers rerenders. React limits renders triggered by useState. Additionally this is quite hard to read. Please look into the useForm Hook for React. I already installed this and used it on the Agenda page to demonstrate how to handle input fields. If you have questions lets look over this together

Reply:
Updated the PR using react-hook-form.
Looks better to me, thanks for the suggestion :)